### ISSUE TITLE
RCP Logging Writing Duplicate or Empty Sample Lines in Log File

### ISSUE THREAD

**Issue by stieg** [autosportlabs/RaceCapture-Pro_firmware#782]
It would appear that RCP is screwing up its logging on 2.10.0.  Specifically the following file was taken using a beta build version of `2.10.0_RC1-77-g1d45062`.  Notice the following:

* Line 153 is a completely empty line:
```csv
,,,,,,,,,,,,,,,,,,,,,,,,,
```
* Lines 315 and 316 are duplicates of one another with missing information on the previous line.
```csv
.16,0.26,-0.36,-63.26,-47.52,-43.75,0,,,0,0.0,,,0,0,0,,,0.0,0.0,0,42.581047,-89.402916,21.58,0.019,784.4,8,3,1.5,1,0.8467,1,0.0483,0.8682,0.0533,2
295859,1469995736320,,0.16,0.26,-0.36,-63.26,-47.52,-43.75,0,,,0,0.0,,,0,0,0,,,0.0,0.0,0,42.581047,-89.402916,21.58,0.019,784.4,8,3,1.5,1,0.8467,1,0.0483,0.8682,0.0533,2
```
* Line 636 and 636 are duplicates of one another:
```csv
,0.21,-0.18,1.04,0.46,-0.88,-0.68,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
308619,1469995749077,,0.21,-0.18,1.04,0.46,-0.88,-0.68,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
```
* Line 798 and 799 are duplicates of one another again.
```csv
16,0.46,0.84,-21.2,-6.86,-7.23,0,0,0,0,0.0,0,0,0,0,0,0,0,0.0,0.0,0,42.581543,-89.403267,15.45,0.101,777.0,9,3,1.6,1,0.8467,1,0.0483,0.4596,0.3733,2
315059,1469995755516,7.74,0.16,0.46,0.84,-21.2,-6.86,-7.23,0,0,0,0,0.0,0,0,0,0,0,0,0,0.0,0.0,0,42.581543,-89.403267,15.45,0.101,777.0,9,3,1.6,1,0.8467,1,0.0483,0.4596,0.3733,2
```
* Empty sample happens again at line 1445.
```csv
,,,,,,,,,,,,,,,,,,,,,,,,,,,,
```

Obviously there are a couple of issues in our FileWriter.  I believe them to be two separate issues, one involving invalid samples (the empty lines) and the other (duplicate lines) involving a failure of the low level driver.

[Heat_3.LOG.txt](https://github.com/autosportlabs/RaceCapture-Pro_firmware/files/392710/Heat_3.LOG.txt)

**Comment by stieg** [comment]
Will move to 2.10.0 radar for now but this will not block 2.10.0 from release.

**Comment by stieg** [comment]
Reproduced this yesterday.  Seems our behavior around write errors does not help.  Going to try maintaining the position within the file and seeking to the correct spot before we write.  With any luck that will eliminate these issues.

**Comment by stieg** [comment]
Fix tracked in PR #794

**Comment by stieg** [comment]
Issue Merged.
